README.md

```diff
@@ -1,23 +1,70 @@
 [![Code Coverage](https://codecov.io/gh/devitocodes/devito/branch/master/graph/badge.svg)](https://codecov.io/gh/devitocodes/devito)
+[![Slack Status](https://img.shields.io/badge/chat-on%20slack-%234A154B)](https://opesci-slackin.now.sh)
+
+[Devito](http://www.devitoproject.org) is a Python package to implement
+optimised tensor computation from high-level symbolic problem definitions.
+Devito builds on [SymPy](http://www.sympy.org/en/index.html) and employs
+automated code generation and just-in-time (JIT) compilation to execute
+optimized computational kernels on multiple computer platforms, including
+CPUs, GPUs, and clusters thereof.
+
+- [About Devito](#about-devito)
+- [Installation](#installation)
+- [Resources](#resources)
+- [Performance](#performance)
+- [Get in touch](#get-in-touch)
+
+## About Devito
+
+Devito provides a symbolic language to implement matrix-free operators, that is
+operators that do not require the explicit creation of a dense or sparse
+matrix. A typical use case is explicit finite difference (FD) methods for
+approximating partial differential equations. For example, a 2D diffusion
+operator may be implemented as follows
+
+```python
+>>> grid = Grid(shape=(10, 10))
+>>> f = TimeFunction(name='f', grid=grid, space_order=2)
+>>> eqn = Eq(f.dt, 0.5 * f.laplace)
+>>> op = Operator(Eq(f.forward, solve(eqn, f.forward)))
+```
 
-[Devito](http://www.devitoproject.org) is a software to
-implement optimised finite difference (FD) computation from
-high-level symbolic problem definitions. Starting from symbolic
-equations defined in [SymPy](http://www.sympy.org/en/index.html),
-Devito employs automated code generation and just-in-time (JIT)
-compilation to execute FD kernels on multiple computer platforms.
-
-## Get in touch
+An `Operator` generates low-level code from an ordered collection of `Eq`. This
+code may also be compiled and executed
 
-If you're using Devito, we would like to hear from you. Whether you
-are facing issues or just trying it out, join the
-[conversation](https://opesci-slackin.now.sh).
+```python
+>>> op(t=timesteps)
+```
 
-## Quickstart
+There is virtually no limit to the type of matrix-free operators that one can
+implement with Devito. The Devito compiler will automatically analyze the input,
+detect and apply optimizations (including single- and multi-node parallelism),
+and eventually generate code with arbitrarily complex loop nests, as required
+by the symbolic specification -- clearly, all this is hidden away to the user.
+
+Key features include:
+
+* A tensor language to express FD operators.
+* Straightforward mechanisms to dynamically adjust the discretizion.
+* Constructs to express sparse operators (e.g., interpolation), classic linear
+  operators (e.g., convolutions), and tensor contractions.
+* Seamless support for adjoint operators.
```

Comment:
Well... if you can implement it by hand.

Reply:
yeah, I'm not saying automated anywhere 

"seamless" refers to the fact that ... "it's just yet another operator"